### PR TITLE
fix: move to async functions where possible

### PR DESCRIPTION
**Pull request by wraithgar**
This refactors the code to use async functions where possible.
It also uses `@npmcli/fs` consistently (it was already a dependency,
just not used).
It also reorders the test files to match their sources in `./lib`.

Tests were not refactored (except where needed to move to `@npmcli/fs`)
in the interest of an easier PR review.

### CODE REVIEW

File: lib/util/fix-owner.js [hunk 1]
```diff
@@ -33,40 +33,38 @@ const getSelf = () => {
...
   getSelf()
   if (self.uid !== 0) {
     // almost certainly can't chown anyway
-    return Promise.resolve()
+    return
   }
 
-  return Promise.resolve(inferOwner(cache)).then((owner) => {
-    const { uid, gid } = owner
+  const { uid, gid } = await inferOwner(cache)
```
Comment (nlf) [inline lib/util/fix-owner.js:48]: the `infer-owner` dependency here can also leverage `@npmcli/fs` in a future refactor pass

File: lib/util/move-file.js [hunk 2]
```diff
@@ -23,47 +18,41 @@ function moveFile (src, dest) {
...
+      throw err
+    }
+  }
+  try {
+    await Promise.all([
+      fs.unlink(src),
+      !isWindows && fs.chmod(dest, '0444'),
+    ])
+  } catch (e) {
+    console.log('unlink error', e.code)
```
Comment (nlf) [inline lib/util/move-file.js:44]: stray `console.log()`

File: lib/util/move-file.js [hunk 3]
```diff
@@ -23,47 +18,41 @@ function moveFile (src, dest) {
...
+    return pinflight('cacache-move-file:' + dest, async () => {
+      await fs.stat(dest).catch((err) => {
+        console.log(err.code)
```
Comment (nlf) [inline lib/util/move-file.js:47]: another stray `console.log()`

File: lib/verify.js [hunk 4]
```diff
@@ -278,16 +243,15 @@ function writeVerifile (cache, opts) {
   const verifile = path.join(cache, '_lastverified')
   opts.log.silly('verify', 'writing verifile to ' + verifile)
   try {
-    return writeFile(verifile, '' + +new Date())
+    return fs.writeFile(verifile, '' + +new Date())
```
Comment (nlf) [inline lib/verify.js:246]:
```suggestion
    return fs.writeFile(verifile, `${Date.now()}`)
```

File: lib/verify.js [hunk 5]
```diff
@@ -278,16 +243,15 @@ function writeVerifile (cache, opts) {
...
 }
 
 module.exports.lastRun = lastRun
 
-function lastRun (cache) {
-  return readFile(path.join(cache, '_lastverified'), 'utf8').then(
-    (data) => new Date(+data)
-  )
+async function lastRun (cache) {
+  const data = await fs.readFile(path.join(cache, '_lastverified'), 'utf8')
```
Comment (nlf) [inline lib/verify.js:255]:
```suggestion
  const data = await fs.readFile(path.join(cache, '_lastverified'), { encoding: 'utf8' })
```

**Review by nlf — CHANGES_REQUESTED** [review]

File: lib/content/read.js [hunk 6]
```diff
@@ -77,16 +70,19 @@ module.exports.readStream = readStream
 function readStream (cache, integrity, opts = {}) {
   const { size } = opts
   const stream = new Pipeline()
-  withContentSri(cache, integrity, (cpath, sri) => {
-    // just lstat to ensure it exists
-    return lstat(cpath).then((stat) => ({ stat, cpath, sri }))
-  }).then(({ stat, cpath, sri }) => {
+  // Set all this up to run on the stream and then just return the stream
+  Promise.resolve().then(async () => {
```
[the remark this answers is not in the document]
Comment (wraithgar) [reply lib/content/read.js:74]: Yeah I fiddled a bit w/ this, this was as good as it got.  The comment *was* supposed to explain it lol.

File: lib/content/read.js [hunk 7]
```diff
@@ -1,42 +1,35 @@
...
-    }
+  if (stat.size > MAX_SINGLE_READ_SIZE) {
+    return readPipeline(cpath, stat.size, sri, new Pipeline()).concat()
+  }
 
-    return readFile(cpath, null).then((data) => {
-      if (!ssri.checkData(data, sri)) {
-        throw integrityError(sri, cpath)
-      }
+  const data = await fs.readFile(cpath, null)
```
[the remark this answers is not in the document]
Comment (wraithgar) [reply lib/content/read.js:27]: the sync method omits it completely, removing second param for consistency

[the remark this answers is not in the document]
Comment (wraithgar) [reply lib/content/read.js:27]: Changed my mind. Putting it in both.  That encoding ensures the result is a buffer. Both are expecting a buffer. Explicit always wins.

**Review by nlf — APPROVED** [review]
this is a great first pass

**Pull request merged**